### DISCUSSION TITLE
canSee can't use $this

### DISCUSSION THREAD

**Discussion by Abather** [laravel/nova-issues#5203] [Q&A] [answered]
- Laravel Version: 9.45.1
- Nova Version: 4.19.5
- PHP Version: 8.1.10
- Database Driver & Version: MySQL
- Operating System and Version: 11
- Browser type and version: Firefox  108.0.1

### Description:
When using the canSee function as describe in the document:
https://nova.laravel.com/docs/4.0/resources/authorization.html#fields

`
use Laravel\Nova\Fields\ID;
use Laravel\Nova\Fields\Text;

/**
 * Get the fields displayed by the resource.
 *
 * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
 * @return array
 */
public function fields(NovaRequest $request)
{
    return [
        ID::make()->sortable(),

        Text::make('Name')
                ->sortable()
                ->canSee(function ($request) {
                    return $request->user()->can('viewProfile', $this);
                }),
    ];
}
`

it fired an error:  
App\Policies\ModeNamePolicy::viwe(): Argument #2 ($modelName) must be of type App\Models\ModeName , App\Nova\ModeName given, called in ...\vendor\laravel\framework\src\Illuminate\Auth\Access\Gate.php on line 798

the function reun correctrly when us the function as:

`
        Text::make('Name')
                ->sortable()
                ->canSee(function ($request) {
                    return $request->user()->can('viewProfile', $this->resource);
                }),
`

**Comment by davidhemphill** [comment, marked as answer]
This is not a bug. You can't pass a Nova resource to a Policy; it must be a Model instance.

**Reply by Abather** [reply]
As I mention the method that written in the document does not work for me, 

`Text::make('Name')
                ->sortable()
                ->canSee(function ($request) {
                    return $request->user()->can('viewProfile', $this);
                }),`

even the other way I was mistaken. 
so what is the correct way to do it when i need to pass model to can?
right now I'm looking for the model that I need then pass it as:

`$supply_contract  = \App\Models\SupplyContract::find($request->resourceId);`

`SellsProductsTotal::make()->canSee(function ($request) use ($supply_contract) {
                return $request->user()->can('viewMetrics', $supply_contract);
            })->onlyOnDetail()`

**Reply by bkuhl** [reply]
You would need to do this:

```
Text::make('Name') ->sortable() ->canSee(function ($request) {
    return $request->user()->can('viewProfile', $this->model());
}),
```

`$this` is the resource, `$this->model()` will provide the model if on the detail view of the page, but an empty model if on the index page.
